src/libraries/System.Drawing.Common/src/System/Drawing/Bitmap.Windows.COMWrappers.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Drawing.Internal;
+using System.Runtime.InteropServices;
+using Gdip = System.Drawing.SafeNativeMethods.Gdip;
+
+namespace System.Drawing
+{
+    public sealed partial class Bitmap
+    {
+        private static unsafe IntPtr CreateGdipBitmapFromStream(GPStream stream, bool useIcm)
+        {
+            DrawingComWrappers.Instance.GetIStreamInterfaces(stream, out IntPtr streamWrapperPtr, out IntPtr streamPtr);
```

Comment:
This seems like it became a very common pattern. How about returning a `struct` that implements `IDisposable` and has fields on it?

Reply:
Or follow @jkotas's suggestion &ndash; https://github.com/dotnet/runtime/pull/54884#discussion_r660193024